docs/docs/general/options.md

```diff
@@ -37,14 +37,52 @@ color: [
 ## Option Context
 
 The option context is used to give contextual information when resolving options and currently only applies to [scriptable options](#scriptable-options).
+The object is preserved, so it can be used to store and pass information between calls.
+
+There are multiple levels of context objects:
+
+- `chart`
+  - `dataset`
+    - `data`
+  - `scale`
+    - `tick`
+
+Each level inherits its parent(s) and any contextual information stored in the parent is available through the child.
 
 The context object contains the following properties:
 
+### chart
+
 - `chart`: the associated chart
-- `dataPoint`: the parsed data values for the given `dataIndex` and `datasetIndex`
-- `dataIndex`: index of the current data
+
+### dataset
+
+In addition to [chart](#chart)
+
+- `active`: true if element is active (hovered)
 - `dataset`: dataset at index `datasetIndex`
 - `datasetIndex`: index of the current dataset
+- `index`: getter for `datasetIndex`
+
+### data
+
+In addition to [dataset](#dataset)
+
 - `active`: true if element is active (hovered)
+- `dataIndex`: index of the current data
+- `dataPoint`: the parsed data values for the given `dataIndex` and `datasetIndex`
+- `element`: the element (point, arc, bar, etc.) for this data
+- `index`: getter for `dataIndex`
```

Comment:
Does this override `index` from the dataset level?

Reply:
Yes it does. You'd use `datasetIndex` when you need to always have that.